src/bench/wallet_create_tx.cpp

```diff
@@ -132,11 +132,52 @@ static void WalletCreateTx(benchmark::Bench& bench, const OutputType output_type
     });
 }
 
+static void AvailableCoins(benchmark::Bench& bench, const std::vector<OutputType>& output_type)
+{
+    const auto test_setup = MakeNoLogFileContext<const TestingSetup>();
+    CWallet wallet{test_setup->m_node.chain.get(), "", gArgs, CreateMockWalletDatabase()};
+    {
+        LOCK(wallet.cs_wallet);
+        wallet.SetWalletFlag(WALLET_FLAG_DESCRIPTORS);
+        wallet.SetupDescriptorScriptPubKeyMans();
+        if (wallet.LoadWallet() != DBErrors::LOAD_OK) assert(false);
+    }
+
+    // Generate destinations
+    std::vector<CScript> dest_wallet;
+    for (auto type : output_type) {
+        dest_wallet.emplace_back(GetScriptForDestination(getNewDestination(wallet, type)));
+    }
+
+    // Generate chain; each coinbase will have two outputs to fill-up the wallet
+    const auto& params = Params();
+    unsigned int chain_size = 1000;
+    for (unsigned int i = 0; i < chain_size / dest_wallet.size(); ++i) {
+        for (const auto& dest : dest_wallet) {
+            generateFakeBlock(params, test_setup->m_node, wallet, dest);
+        }
+    }
+
+    // Check available balance
+    auto bal = wallet::GetAvailableBalance(wallet); // Cache
+    assert(bal == 50 * COIN * (chain_size - COINBASE_MATURITY));
+
+    bench.epochIterations(2).run([&] {
+        LOCK(wallet.cs_wallet);
+        const auto& res = wallet::AvailableCoins(wallet);
+        std::cout << res.All().size() << std::endl;
```

Comment:
I think this is a debugging leftover?

Reply:
it's all about perspective, could be a feature too.

fixed, thanks.